### DISCUSSION TITLE
Enable  "Ask for admin privileges once for each batch of operations" remotely by command/script?

### DISCUSSION THREAD

**Discussion by YuriMri** [Devolutions/UniGetUI#2718] [Q&A] [answered]
Hi.
Is there any option to enable setting "Ask for admin privileges once for each batch of operations" remotely?
I am deploying UniGetUI as software updater for users in our company and managed to silently deploy it without user interaction (installed for all users).
Then I set registry entry for all users making UniGetUI run after login and last part I would like to achieve is that when users see couple of software packages pending, hit one "Update all", accept UAC and have it all installed while they go for lunch or meeting or whatever so when they come back, all will be updated.

I found this setting "Ask for admin privileges once for each batch of operations" that basically does the trick, but I'd like to set up whole UniGet so users don't have to switch it on manually after UniGet installation is finished.

Is there a way of enabling it or achieving my goal different way?
Thank you

**Comment by marticliment** [comment, marked as answer]
Settings deployment is not a thing yet, but if you create an empty file under `%localappdata%\UniGetUI` named `DoCacheAdminRightsForBatches` and then restart UniGetUI the feature should get enabled

**Comment by YuriMri** [comment]
@marticliment Thank you. Exactly what I needed for this job (clever design by the way and easy to manage). 

I noticed that changing the settings in the program will create or remove files from the "%localappdata%\UniGetUI" directory. Not tested all settings so want to ask if all settings works this way or are there any exceptions?
Just wondering what can I tweak with this.

**Reply by Lockszmith-GH** [reply]
I searched for `` in the code, and saw that it was being called by [Setting.Get](https://github.com/marticliment/UniGetUI/blob/dc1a177d8ba4dbc09bed0039e62d23bbe4a405c5/src/UniGetUI.Core.Settings/SettingsEngine.cs#L33), which is defined as:

```c#
        public static bool Get(string setting, bool invert = false)
        {
            return File.Exists(Path.Join(CoreData.UniGetUIDataDirectory, setting)) ^ invert;
        }
```

So it seems that is the design choice **\@marticliement** made was that all settings are saved like that.

From the code view, GitHub should present you with a Symbol legend side bar, select the Get method, and find all the references (currently there are 13).

<details> <summary> click to view screenshot </summary>

![image](https://github.com/user-attachments/assets/436deccd-186c-47cb-bccb-f66b2d8d268c)

</details>

Out of the 13, I managed to identify these:

* `DisableSystemTray`
* `DisableNotifications`
* `DisableUpdatesNotifications`
* `DisableErrorNotifications`
* `DisableSuccessNotifications`
* `DisableProgressNotifications`
* `IconDataBaseURL`
* `UseUserGSudo`
* `DisableAutoUpdateWingetUI`
* `DisableApi`
* `SkippedInstalling{dependency.Name}`

But clearly there are more (I mean, `DoCacheAdminRightsForBatches` doesn't show up), so I also searched for [Settings.Get](https://github.com/search?q=repo%3Amarticliment%2FUniGetUI+Settings.Get&type=code) - that brought a lot more results.

**Reply by marticliment** [reply]
> So it seems that is the design choice **@marticliement** made was that all settings are saved like that.

Yes, but this will be changed in a future, since this was a terrible choice from my side (very slow on slower file systems). 

But yes, if you enable a setting you will see a file appear on `%localappdata%\UniGetUI`, and this is the name of the setting
